### PR TITLE
[DO NOT MERGE] Added classes to watch for is, ittag, and ConfigMaps

### PR DESCRIPTION
**Pull request by oatmealraisin**
Added classes to watch for ImageStreams, ImageStreamTags, and ConfigMaps in order to create PodTemplates from them.

@gabemontero ptal?

### CODE REVIEW

**Comment by oatmealraisin** [comment]
```
apiVersion: v1
items:
- apiVersion: v1
  data:
    asdf: |-
      <org.csanchez.jenkins.plugins.kubernetes.PodTemplate>
        <inheritFrom></inheritFrom>
        <name>asdf</name>
        <instanceCap>2147483647</instanceCap>
        <idleMinutes>0</idleMinutes>
        <label>maven</label>
        <serviceAccount>jenkins</serviceAccount>
        <nodeSelector></nodeSelector>
        <volumes/>
        <containers>
          <org.csanchez.jenkins.plugins.kubernetes.ContainerTemplate>
            <name>jnlp</name>
            <image>openshift/jenkins-slave-maven-centos7</image>
            <privileged>false</privileged>
            <alwaysPullImage>false</alwaysPullImage>
            <workingDir>/tmp</workingDir>
            <command></command>
            <args>${computer.jnlpmac} ${computer.name}</args>
            <ttyEnabled>false</ttyEnabled>
            <resourceRequestCpu></resourceRequestCpu>
            <resourceRequestMemory></resourceRequestMemory>
            <resourceLimitCpu></resourceLimitCpu>
            <resourceLimitMemory></resourceLimitMemory>
            <envVars/>
          </org.csanchez.jenkins.plugins.kubernetes.ContainerTemplate>
        </containers>
        <envVars/>
        <annotations/>
        <imagePullSecrets/>
        <nodeProperties/>
      </org.csanchez.jenkins.plugins.kubernetes.PodTemplate>
    fdsa: |-
      <org.csanchez.jenkins.plugins.kubernetes.PodTemplate>
        <inheritFrom></inheritFrom>
        <name>fdsa</name>
        <instanceCap>2147483647</instanceCap>
        <idleMinutes>0</idleMinutes>
        <label>maven</label>
        <serviceAccount>jenkins</serviceAccount>
        <nodeSelector></nodeSelector>
        <volumes/>
        <containers>
          <org.csanchez.jenkins.plugins.kubernetes.ContainerTemplate>
            <name>jnlp</name>
            <image>openshift/jenkins-slave-maven-centos7</image>
            <privileged>false</privileged>
            <alwaysPullImage>false</alwaysPullImage>
            <workingDir>/tmp</workingDir>
            <command></command>
            <args>${computer.jnlpmac} ${computer.name}</args>
            <ttyEnabled>false</ttyEnabled>
            <resourceRequestCpu></resourceRequestCpu>
            <resourceRequestMemory></resourceRequestMemory>
            <resourceLimitCpu></resourceLimitCpu>
            <resourceLimitMemory></resourceLimitMemory>
            <envVars/>
          </org.csanchez.jenkins.plugins.kubernetes.ContainerTemplate>
        </containers>
        <envVars/>
        <annotations/>
        <imagePullSecrets/>
        <nodeProperties/>
      </org.csanchez.jenkins.plugins.kubernetes.PodTemplate>
  kind: ConfigMap
  metadata:
    annotations:
      openshift.io/generated-by: OpenShiftNewApp
    creationTimestamp: 2017-05-22T21:16:22Z
    labels:
      app: jenkins-slave-builder
      template: jenkins-slave-builder
    name: jenkins-slave
    namespace: myproject
    resourceVersion: "2277"
    selfLink: /api/v1/namespaces/myproject/configmaps/jenkins-slave
    uid: e7b3b766-3f33-11e7-a708-507b9daca183
kind: List
metadata: {}
resourceVersion: ""
selfLink: ""
```

**Comment by oatmealraisin** [comment]
@gabemontero They're just the Maven template, but with different names

File: src/main/java/io/fabric8/jenkins/openshiftsync/GlobalPluginConfiguration.java [hunk 1]
```diff
@@ -185,6 +191,12 @@ protected void doRun() throws Exception {
           buildConfigWatcher.start();
           buildWatcher = new BuildWatcher(namespaces);
           buildWatcher.start();
+          configMapWatcher = new ConfigMapWatcher(namespaces);
+          configMapWatcher.start();
+          imageStreamWatcher = new ImageStreamWatcher(namespaces);
+          imageStreamWatcher.start();
+          //imageStreamTagWatcher = new ImageStreamTagWatcher(namespaces);
+          //imageStreamTagWatcher.start();
```
Comment (oatmealraisin) [inline src/main/java/io/fabric8/jenkins/openshiftsync/GlobalPluginConfiguration.java]: I'm working on it right now, we definitely need this feature for the card.

Something is printing logs every now and again with it, which can flood the logger after a while. It works, though, so we can uncomment it :)

File: src/main/java/io/fabric8/jenkins/openshiftsync/ImageStreamTagWatcher.java [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    private synchronized void onInitialImageStreamTag(ImageStreamTagList imageStreamTags) {
+        if(trackedImageStreamTags == null) {
+            trackedImageStreamTags = new ConcurrentHashMap<>(imageStreamTags.getItems().size());
+        }
+        List<ImageStreamTag> items = imageStreamTags.getItems();
+        if (items != null) {
+            for (ImageStreamTag imageStreamTag : items) {
+                try {
+                    if(isSlaveImage(imageStreamTag) && !trackedImageStreamTags.containsKey(imageStreamTag.getMetadata().getUid())) {
+                        logger.info("GREPME: onInitialImageStreamTag FOUND SLAVE");
```
Comment (oatmealraisin) [inline src/main/java/io/fabric8/jenkins/openshiftsync/ImageStreamTagWatcher.java]: Sorry, I had these in for me. They've been removed

**Comment by gabemontero** [comment]
hey @oatmealraisin - in case it got lost in the IRC shuffle, if that exception you mentioned you were shooting is still in play before you stop on  this one, please post any exception stack traces, steps to reproduce, etc. that you have - thanks !

**Comment by gabemontero** [comment]
[(5) Better jenkins slave template configuration [jenkins_integration]](https://trello.com/c/hS9lzo1v/1202-5-better-jenkins-slave-template-configuration-jenkinsintegration)

**Pull request closed without merging**